hawc/apps/summary/migrations/0059_move_data_pivot.py

```diff
@@ -0,0 +1,186 @@
+            last_updated=dp.last_updated,
+        )
+        visualization.save()
+
+        mapping.append(("dpq", dp.id, visualization.id))
+
+        visualization.created = dp.created
+        visualization.last_updated = dp.last_updated
+        dps.append(visualization)
+
+        for label in LabeledItem.objects.filter(object_id=dp.id, content_type_id=dpq_ct):
+            new_label = LabeledItem.objects.create(
+                object_id=visualization.id, content_type_id=dpq_ct, label_id=label.label_id
+            )
+            mapping.append(("label", label.id, new_label.id))
+
+    for dp in DataPivotUpload.objects.all():
+        new_dp_id = (
+            f"{dp.slug}-dp"
+            if Visual.objects.filter(assessment_id=dp.assessment_id, slug=dp.slug).exists()
+            else dp.slug
+        )
+        new_ds_name = (
+            f"{dp.title} dp"
+            if Dataset.objects.filter(assessment_id=dp.assessment_id, name=dp.title).exists()
+            else dp.title
+        )
+
+        if settings.DEBUG:
+            df = pd.DataFrame(data=[[1, 2], [3, 4]], columns=("a", "b"))
+            data = BytesIO(b"debugging")
+        else:
+            worksheet_name = dp.worksheet_name or 0
+            df = pd.read_excel(dp.excel_file.file, sheet_name=worksheet_name)
+            data = BytesIO(dp.excel_file.file.read())
+
+        dataset = create_dataset(
+            apps,
+            assessment_id=dp.assessment_id,
+            name=new_ds_name,
+            description=dp.caption,
+            data=data,
+            df=df,
+            data_name=Path(dp.excel_file.name).name,
+            published=True,
+            excel_worksheet_name=dp.worksheet_name,
+            notes="",
+        )
+        visualization = Visual(
+            title=dp.title,
+            slug=new_dp_id,
+            assessment=dp.assessment,
+            visual_type=VisualType.DATA_PIVOT_FILE,
+            evidence_type=StudyType.OTHER,
+            settings=dp.settings,
+            caption=dp.caption,
+            published=dp.published,
+            dataset=dataset,
+            prefilters={},
+            dp_id=dp.id,
+            dp_slug=dp.slug,
+            created=dp.created,
+            last_updated=dp.last_updated,
+        )
+        visualization.save()
+
+        mapping.append(("dpu", dp.id, visualization.id))
+
+        visualization.created = dp.created
+        visualization.last_updated = dp.last_updated
+        dps.append(visualization)
+
+        for label in LabeledItem.objects.filter(object_id=dp.id, content_type_id=dpu_ct):
+            new_label = LabeledItem.objects.create(
+                object_id=visualization.id, content_type_id=dpu_ct, label_id=label.label_id
+            )
+            mapping.append(("label", label.id, new_label.id))
+
+    Visual.objects.bulk_update(dps, fields=("created", "last_updated"))
+
+    pd.DataFrame(data=mapping, columns="model|old_id|new_id".split("|")).to_csv(
```

Comment:
Not sure if this excel was generated just for debugging purposes, but if you intend to keep this code here consider adding the generated excel file to `gitignore`.